app/controllers/PasskeyController.scala

```diff
@@ -143,33 +144,48 @@ class PasskeyController(
     )
   }
 
-  /** Deletes a passkey for a user */
-  def deletePasskey: Action[Map[String, Seq[String]]] = authAction(
-    parse.formUrlEncoded
-  ) { implicit request =>
-    apiResponse(
-      for {
-        passkeyId <- request.body.get("passkeyId") match {
-          case Some(values) => Success(values.head)
-          case None =>
-            Failure(
-              JanusException.missingFieldInRequest(request.user, "passkeyId")
+  /** Deletes a passkey from the user's account */
+  def deletePasskey(passkeyId: String): Action[Unit] = authAction(parse.empty) {
+    implicit request: UserIdentityRequest[Unit] =>
+      apiResponse(
+        for {
+          // Look up the passkey before deleting to include the name in the success message
+          queryResponse <- PasskeyDB.loadCredentials(request.user)
+          passkeys = PasskeyDB.extractMetadata(queryResponse)
+          passkeyName <- passkeys
+            .find(_.id == passkeyId)
+            .map(_.name)
+            .toRight(
+              JanusException(
+                userMessage = "Passkey not found",
+                engineerMessage =
+                  s"Passkey with ID $passkeyId not found for user ${request.user.username}",
+                httpCode = NOT_FOUND,
+                causedBy = Some(
```

Comment:
Generally the different types of `JanusException` are in `models` so that their details don't clog up the controller code and they can easily be compared with each other and to avoid repetition.